libs/core/hardware/include/hpx/hardware/timestamp/cuda.hpp

```diff
@@ -0,0 +1,24 @@
+////////////////////////////////////////////////////////////////////////////////
+//  Copyright (c) 2012 Thomas Heller
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <cuda.h>
+
+#include <hpx/config.hpp>
+
+namespace hpx { namespace util { namespace hardware {
+
+    HPX_HOST_DEVICE std::uint64_t timestamp()
```

Comment:
This `HPX_HOST_DEVICE` doesn't look right, it should just be `HPX_DEVICE`, but then I guess it's related to the comment above?

Reply:
In theory, it should work for both but HPX_DEVICE is more appropriate here. Let me update the whole function again so that it caters to the problem I described above.